e2e_test/source/cdc_inline/alter/cdc_table_alter.slt

```diff
@@ -0,0 +1,223 @@
+control substitution on
+
+system ok
+mysql -u root --protocol='tcp' -e "
+    SET GLOBAL time_zone = '+00:00';
+"
+
+system ok
+mysql -u root --protocol='tcp' -e "
+  DROP DATABASE IF EXISTS testdb1;
+  CREATE DATABASE testdb1;
+  USE testdb1;
+  CREATE TABLE products (
+      id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+      name VARCHAR(255) NOT NULL,
+      description VARCHAR(512)
+  );
+  ALTER TABLE products AUTO_INCREMENT = 101;
+  INSERT INTO products
+  VALUES (default,'scooter','Small 2-wheel scooter'),
+         (default,'car battery','12V car battery'),
+         (default,'12-pack drill','12-pack of drill bits with sizes ranging from #40 to #3'),
+         (default,'hammer','12oz carpenter s hammer'),
+         (default,'hammer','14oz carpenter s hammer'),
+         (default,'hammer','16oz carpenter s hammer'),
+         (default,'rocks','box of assorted rocks'),
+         (default,'jacket','water resistent black wind breaker'),
+         (default,'spare tire','24 inch spare tire');
+  CREATE TABLE orders (
+      order_id INTEGER NOT NULL AUTO_INCREMENT PRIMARY KEY,
+      order_date DATETIME NOT NULL,
+      customer_name VARCHAR(255) NOT NULL,
+      price DECIMAL(10, 5) NOT NULL,
+      product_id INTEGER NOT NULL,
+      order_status BOOLEAN NOT NULL
+  ) AUTO_INCREMENT = 10001;
+  INSERT INTO orders
+  VALUES (default, '2020-07-30 10:08:22', 'Jark', 50.50, 102, false),
+         (default, '2020-07-30 10:11:09', 'Sally', 15.00, 105, false),
+         (default, '2020-07-30 12:00:30', 'Edward', 25.25, 106, false);
+"
+
+statement ok
+create source mysql_source with (
+  connector = 'mysql-cdc',
+  hostname = '${MYSQL_HOST:localhost}',
+  port = '${MYSQL_TCP_PORT:8306}',
+  username = 'root',
+  password = '123456',
+  database.name = 'testdb1',
+  server.id = '5185'
```

Comment:
Can we use `${RISEDEV_MYSQL_WITH_OPTIONS_COMMON}`?

----

Seems that similar comments have been made in https://github.com/risingwavelabs/risingwave/pull/16986#discussion_r1630596056.

Reply:
> It seems we have to launch the mysql via `risedev` to let the `RISEDEV_MYSQL_WITH_OPTIONS_COMMON` work, but I use `docker compose` to launch mysql locally.

This is true. However, I would still recommend launching MySQL with RiseDev as it makes it easier for other developers to test locally without needing to follow any other conventions.

Reply:
> but I use docker compose to launch mysql locally.

You CAN still use your own setup, by using `user-managed`.

Reply:
Any reason why you don't use the mysql started by `risedev`? Have you tried it and met any problems?

Reply:
> Basically, I don't get the point to use this placeholder, which only eliminate three options. It is not a big deal that we repeat these options in the test case. Because the host and port will be substituted with MYSQL_ env variables eventually.

Yes, `WITH_OPTIONS_COMMON` is only a little style problem. It's not a big deal.

What matters is the `RISEDEV_` part. i.e., `risedev` reads the config in `risedev.yml`, and them set these env vars, and load them when using `risedev slt`.


So:
- **You don't need to manually set `MYSQL_` env vars manually**.
- When you change the configs, the env vars are also updated automatically.

> So it is unnecessary to resort to risedev to do the thing. I can stop and start one or more db services with clicks, no need to edit the yaml.

First, I think the point above already answers the problem partly: You still need to set the env vars. I think the automatic env var handling is a huge DX improvement.

I can understand that when you already have the setup working, it might seems unnecessary for you to change. However, for other developers, the new workflow is much more easy to adopt.

Reply:
> I can stop and start one or more db services with clicks, no need to edit the yaml.

You still uses `risedev` to start RisingWave. So relying on `risedev` can also save the click.. Why not?

---

Anyway, I encourage you to just give it a try. I believe basically no effort is needed to change the workflow and the DX improvement is gained for free.

If it doesn't work well, please share some feedbacks about what is not good enough, so that we can keep improving the DX for all developers.

If the reasons above are still not convincing enough to you, please also tell me why, so that I can understand it better. 

Finally, if you just don't want to try it, it's also OK, but please use the suggested style for the benefits of others. For it to work in your machine, I think it's just some additional env vars.

Reply:
Sorry for missing this comment.

Just to double confirm: Did you run with `risedev slt`? And what's the errors you found?

Perhaps you can try this first:

```
system ok
echo $PGPORT
----
placeholder
```

Then running `risedev slt` will return errors "result mismatch", and shows what's the output of the `echo` command, i.e., the value of `PGPORT`.

Reply:
> Sorry for missing this comment.

Wait, why this reply is "2 days ago", but I really just received it just now. 😂

Reply:
> > Sorry for missing this comment.
> 
> Wait, why this reply is "2 days ago", but I really just received it just now. 😂

Maybe due to I forget to submit the pending comment. 😄